src/Beacon.jsx

```diff
@@ -152,140 +84,78 @@ export class Beacon extends React.Component {
-      </Portal>
+      <TetherComponent attachment="middle center" constraints={[{ to: 'scrollParent' }]}>
+        {React.cloneElement(this.props.children, { ref: 'tooltipTarget' })}
+        <div>
+          <tour-beacon ref="beacon" onClick={::this.showTooltip}><span /></tour-beacon>
+        </div>
+      </TetherComponent>
     );
   }
 
-  renderTooltip(position, children) {
-    const { left, top, className } = this.getTooltipCoordinates(position);
-    if (className && this.state.appRoot) {
+  renderTooltip() {
+    const { appRoot, appRootClassName, tooltipActive } = this.state;
+
+    if (appRoot) {
       const oldClone = document.getElementById(TARGET_CLONE_ID);
-      if (!this.state.tooltipActive && oldClone) {
+      if (!tooltipActive && oldClone) {
         oldClone.className = 'tour-clone';
         oldClone.addEventListener('transitionend', () => {
           oldClone.parentNode.removeChild(oldClone);
+          // Remember that we hid the tooltip so we don't use the `tour-out` class anymore
+          this.setState({ tooltipHidden: true });
         }
         , false);
       }
-      // If we have a `className` (i.e. we have the tooltip size and are rendering onscreen)
-      // and the user specified an app root using the `TOOLTIP_OVERLAY_CLASS`
+      // If the user specified an app root using the `TOOLTIP_OVERLAY_CLASS`
       // then we fade out the background and highlight the target element of the tooltip.
-      const beacons = Array.prototype.slice.call(document.getElementsByTagName('tour-beacon'));
-      if (this.state.tooltipActive) {
+      if (tooltipActive) {
         if (!oldClone) {
-          this.state.appRoot.className = `${TOOLTIP_FADED_CLASS} ${this.state.appRootClassName}`;
+          appRoot.className = `${TOOLTIP_FADED_CLASS} ${appRootClassName}`;
           const targetClone = this.getTargetClone();
           document.body.appendChild(targetClone);
-          // Hide beacons while tooltip is visible
-          beacons.forEach(beacon => beacon.style.display = 'none');
         }
       } else {
-        this.state.appRoot.className = this.state.appRootClassName;
-        // Show beacons again
-        beacons.forEach(beacon => beacon.style.display = 'block');
+        appRoot.className = appRootClassName;
       }
     }
-    const style = {
-      position: 'absolute',
-      left: `${left}px`,
-      top: `${top}px`
-    };
+
+    const tooltipClass = tooltipActive ? 'tour-in' : !this.state.tooltipHidden && 'tour-out';
     return (
-      <div>
-        <Portal isOpened onOpen={this.updateState.bind(this, this.props, this.state)}>
-          <tour-tooltip ref="tooltip" class={className} style={style}>
-            {children}
-          </tour-tooltip>
-        </Portal>
-      </div>
+      <TetherComponent
+        attachment="top left"
+        constraints={[
+          {
+            to: 'scrollParent',
+            attachment: 'together'
+          }
+        ]}
+      >
```

Comment:
These `constraints` make the tooltip flip around to the other side when it moves out of the scroll area.